stacks_queues/queue.py

```diff
@@ -15,47 +15,80 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException("Queue is full")
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException("Queue is empty")
+
+        dequeued = self.store[self.front]
+        self.store[self.front] = None
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+
+        return dequeued
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.empty():
+            return None
+        else:
+            return self.store[self.front]
+        
         
-
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        return not bool(self.size)
 
     def __str__(self):
```

Comment:
👍 

When using slicing to extract a portion of a list, consider using `itertools.islice`, which avoids the copy regular slicing performs (though at the potential cost a slightly more iterations). To avoid either, we can manage our our index to copy values from the buffer to the value list.